src/help/help_topic_generators.cpp

```diff
@@ -646,149 +650,175 @@ std::string unit_topic_generator::operator()() const {
 			}
 
 			if (info.union_type().size() == 1 && info.union_type()[0] == info.number() && info.is_nonnull()) {
-				std::vector<item> row;
-				const std::string& name = info.name();
-				const std::string& id = info.id();
-				const int views = movement_type.vision_cost(terrain);
-				const int jams  = movement_type.jamming_cost(terrain);
-
-				bool high_res = false;
-				const std::string tc_base = high_res ? "images/buttons/icon-base-32.png" : "images/buttons/icon-base-16.png";
-				const std::string terrain_image = "icons/terrain/terrain_type_" + id + (high_res ? "_30.png" : ".png");
-
-				const std::string final_image = tc_base + "~RC(magenta>" + id + ")~BLIT(" + terrain_image + ")";
-
-				row.emplace_back("<img>src='" + final_image + "'</img> " +
-						make_link(name, "..terrain_" + id),
-					font::line_width(name, normal_font_size) + (high_res ? 32 : 16) );
-
-				//defense  -  range: +10 % .. +70 %
-				const int defense = 100 - movement_type.defense_modifier(terrain);
-				std::string color;
-				if (defense <= 10) {
-					color = "red";
-				} else if (defense <= 30) {
-					color = "yellow";
-				} else if (defense <= 50) {
-					color = "white";
+				terrain_moves.push_back(std::make_tuple(info.name(), // 0 name (t_string)
+					info.id(),  // 1 id (t_string)
+					100 - movement_type.defense_modifier(terrain), // 2 defense (int)
+					moves, // 3 movement_cost (int)
+					movement_type.vision_cost(terrain), // 4 vision_cost (int)
+					movement_type.jamming_cost(terrain), // 5 jamming_cost (int)
+					movement_type.get_defense().capped(terrain))); // 6 defense_cap (bool)
+			}
+		}
+
+		// SORT HERE
+		auto movement_compare_ascending = [](std::tuple<t_string, t_string, int, int, int, int, bool> a,
+			std::tuple<t_string, t_string, int, int, int, int, bool> b)
+		{
+			return std::get<0>(a) < std::get<0>(b);
+		};
+
+		std::sort(terrain_moves.begin(), terrain_moves.end(), movement_compare_ascending); 
+
+		for(unsigned int i = 0; i < terrain_moves.size(); i++)
```

Comment:
Please use range-based for loop.

```cpp
for(const terrain_move& m : terrain_moves)
```

(In this example, `terrain_move` is the name of the custom struct I recommend above.)